sdk/test-utils/recorder/src/utils/msalAuth.node.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { env } from ".";
+
+export type NockType = typeof import("nock");
+let nock: NockType;
+
+/**
+ *  msal auth requests where an access_token is returned are dynamic
+ *  - client_request_id is generated everytime we make a new request which makes the requestBody dynamic
+ *  - requestBody also has more properties such as the type/make of the OS which makes the requests different in different machines
+ *
+ *  This method provides the fake token during playback by not matching the request body at all
+ */
+export function mockMsalAuth(importNock: NockType, plugin: (() => void) | undefined) {
```

Comment:
Ah, I meant to call `plugin` in line 21.

Reply:
Currently, identity SDK tests import the method `pluginForIdentitySDK` which is exported from the recorder to pass on.

I kept the param as a function just in case identity wants to pass a new function to handle more things in future (if at all needed) when the default one provided is not enough.
If not for identity, we won't be using this param.

Reply:
Good catch!